Инкремент №14

PaulYakow opened the pull request.

PaulYakow committed:
+ Добавлен сбор дополнительных метрик

PaulYakow committed:
+ Исправлен go.sum

PaulYakow committed:
+ Исправлено название функции getCPUInfo и поле получения свободной памяти (Free -> Available)

PaulYakow committed:
+ Исправлено обратно поле получения свободной памяти (на Free) и невнимательность в использовании функции обновления (UpdateValue)

PaulYakow committed:
+ controller/server -> добавлена реализация с использованием библиотеки gin, а также middleware при обработке запросов (v2)
+ scheduler перенесён в корень controller/server
+ controller/client -> у collector'а контекст перенесён из структуры в параметры метода Run

PaulYakow committed:
+ controller/server (v2) -> исправление маршрутов (статус 307), исправление имён констант и методов (vet test)

PaulYakow committed:
+ controller/server (v2) -> исправлена недоработка при отправке сохранённых значений (route "value/")

rekby commented on internal/usecase/repo/client_memory.go:
```diff
@@ -5,32 +5,38 @@ import (
 
 	idx := 0
-	for _, metric := range r.metrics {
+	for _, metric := range repo.metrics {
```
тут можно написать:
for idx, metric := range repo.metrics {
...

тогда не надо отдельно объявлять индекс и помнить об инкрементах

PaulYakow replied:
К сожалению нельзя, т.к. repo.metrics это map[string], а result - slice.

rekby commented on internal/controller/server/v2/root.go:
```diff
@@ -0,0 +1,57 @@
+}
+
+func (r *rootRoutes) listOfMetrics(c *gin.Context) {
+	tmpl := template.Must(template.ParseFiles(templateName))
```
Тут я бы использовал сохранение tmpl внутрь экземпляра r если не требуется менять шаблон "на ходу" - без перезапуска приложения - чтобы не парсить его на каждый запрос.

Содержимое tmpl не зависит от настроек, количества потоков, оно не меняется и т.п. - т.е. зависит исключительно от содержимого шаблона и если бы синтаксис позволял, то его можно было бы объявить константой на уровне пакета.

Заодно и парситься шаблон будет один раз - при старте приложения, а не на каждый запрос.

PaulYakow committed:
Устранение замечаний код-ревью
+ controller/server -> сохранение шаблона перенесено в экземпляр корневого роута

PaulYakow committed:
Устранение замечаний код-ревью
+ controller/server (v1) -> исправлен тестовый файл
